scorecard/profile_data.py

```diff
@@ -236,19 +241,79 @@ def revenue_breakdown(self):
                         'item': item,
                         'amount': amount,
                         'percent': percent(amount, total),
-                        'year': year_name,
+                        'date': year_name,
                     })
                 if total and subtotal and (total != subtotal):
                     values.append({
                         'item': 'Other',
                         'amount': total - subtotal,
                         'percent': percent(total - subtotal, total),
-                        'year': year_name,
+                        'date': year_name,
                     })
             except KeyError:
                 continue
         return {'values': values}
 
+    def current_ratio(self):
+        values = []
+        results = self.results['in_year_bsheet']
+        year_key = lambda r: r['financial_year_end.year']
+        month_key = lambda r: r['financial_period.period']
+        year_sorted = sorted(results, key=year_key)
+        year_sorted.reverse()
+        quarters = {}
+        latest_quarter = None
+        for year, yeargroup in groupby(year_sorted, year_key):
+            month_sorted =  sorted(yeargroup, key=month_key)
+            month_sorted.reverse()
+            # Loop over months that exist and add their values to quarters
+            for month, monthgroup in groupby(month_sorted, key=month_key):
+                monthitems = list(monthgroup)
+                quarter_idx = ((month - 1) / 3) + 1
+                quarter_key = (year, quarter_idx)
+                try:
+                    # Rely on index out of range for missing values to skip month if one's missing
+                    assets = [m['amount.sum'] for m in monthitems if m['item.code'] == '2150'][0]
+                    liabilities = [m['amount.sum'] for m in monthitems if m['item.code'] == '1600'][0]
+
+                    if quarter_key not in quarters:
```

Comment:
that's right. The month is included in the data passed to the frontend in case we want t say "and in this month we actually used month 8 and not month 9